### PR TITLE
Update the `k8s_sat` node attestor to work with projected tokens also

### LINKED ISSUE #2419: `k8s_sat` node attestor: make it work with the k8s BoundServiceAccountTokenVolume feature
Starting with Kubernetes 1.21, the [BoundServiceAccountTokenVolume](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume) feature is enabled by default, which effectively means that a projected service account token is used instead of a regular service account token.

We need to:
- Make the changes needed to correctly parse the namespace and service account.
- Probably suggest the user to either switch to the `k8s_psat` node attestor or use the Token Review API by enabling the `use_token_review_api_validation` configurable if possible.

### PR DESCRIPTION
**Pull request by amartinezfayo**
Starting with Kubernetes 1.21, the [BoundServiceAccountTokenVolume](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#bound-service-account-token-volume) feature is enabled by default, which means that a projected service account token is used instead of a regular service account token.
This PR updates the `k8s_sat` node attestor to work with projected service account tokens also.

The current implementation validates the token issuer with a specific hardcoded issuer. This validation is not strictly needed by the attestation process and can be problematic in case that a non-default issuer has been defined in the cluster. This PR also updates the logic to remove that validation.

Fixes #2419.

### CODE REVIEW

File: pkg/server/plugin/nodeattestor/k8s/sat/sat.go [hunk 1]
```diff
@@ -187,24 +195,23 @@ func (p *AttestorPlugin) Attest(stream nodeattestorv1.NodeAttestor_AttestServer)
...
-			Issuer: "kubernetes/serviceaccount",
-		}); err != nil {
-			return status.Errorf(codes.InvalidArgument, "unable to validate token claims: %v", err)
-		}
-
-		if claims.Namespace == "" {
-			return status.Error(codes.InvalidArgument, "token missing namespace claim")
+		if !claims.Expiry.Time().IsZero() {
+			// This is an indication that this may be a projected service account token
+			p.log.Warn(`The service account token has an expiration time, which is an indication that may be a projected service account token.
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/k8s/sat/sat.go:200]: How does this message show up in the logging? the  `` literal includes a newline and a bunch of tabs. Should we remove those?

File: pkg/server/plugin/nodeattestor/k8s/sat/sat.go [hunk 2]
```diff
@@ -226,6 +233,36 @@ func (p *AttestorPlugin) Attest(stream nodeattestorv1.NodeAttestor_AttestServer)
...
+// to extract the namespace and service account name
+func (p *AttestorPlugin) getNamesFromClaims(claims *k8s.SATClaims) (namespace string, serviceAccountName string, err error) {
+	if claims.Namespace == "" {
+		if claims.K8s.Namespace == "" {
+			return "", "", errors.New("token missing namespace claim")
+		}
+		namespace = claims.K8s.Namespace
+	} else {
+		if claims.K8s.Namespace != "" {
+			return "", "", errors.New("malformed token: namespace found in two places")
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/k8s/sat/sat.go:246]:
```suggestion
			return "", "", errors.New("malformed token: namespace found in two claims")
```

File: pkg/server/plugin/nodeattestor/k8s/sat/sat.go [hunk 3]
```diff
@@ -226,6 +233,36 @@ func (p *AttestorPlugin) Attest(stream nodeattestorv1.NodeAttestor_AttestServer)
...
+	}
+
+	if claims.ServiceAccountName == "" {
+		if claims.K8s.ServiceAccount.Name == "" {
+			return "", "", errors.New("token missing service account name claim")
+		}
+		serviceAccountName = claims.K8s.ServiceAccount.Name
+	} else {
+		if claims.K8s.ServiceAccount.Name != "" {
+			return "", "", errors.New("malformed token: service account name found in two places")
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/k8s/sat/sat.go:258]:
```suggestion
			return "", "", errors.New("malformed token: service account name found in two claims")
```

**Review by azdagron — COMMENTED** [review]
Thanks, @amartinezfayo ! Just a few small comments...

File: pkg/server/plugin/nodeattestor/k8s/sat/sat.go (hunk 1 above)
Comment (amartinezfayo) [reply pkg/server/plugin/nodeattestor/k8s/sat/sat.go:200]: I'll remove those special characters.

**Review by azdagron — APPROVED** [review]
\o/

**Pull request merged**